openstf/model/model_creator.py

```diff
@@ -6,21 +6,31 @@
 from lightgbm import LGBMRegressor
 
 from openstf.enums import MLModelType
+from openstf.model.xgb_quantile import XgbQuantile
 
 
 class ModelCreator:
+
     """Factory object for creating model trainer objects"""
 
     # Set object mapping
     MODEL_TRAINER_CONSTRUCTORS = {
         MLModelType.XGB: XGBRegressor,
         MLModelType.LGB: LGBMRegressor,
+        MLModelType.XGB_QUANTILE: XgbQuantile,
     }
 
     @staticmethod
-    def create_model(model_type: MLModelType) -> RegressorMixin:
+    def create_model(pj: dict) -> RegressorMixin:
         # check if model type is valid
-        if model_type not in [k.value for k in ModelCreator.MODEL_TRAINER_CONSTRUCTORS]:
-            raise KeyError(f"Unknown model type: {model_type}")
+        if pj["model"] not in [
+            k.value for k in ModelCreator.MODEL_TRAINER_CONSTRUCTORS
+        ]:
+            raise KeyError(f"Unknown model type: {pj['model']}")
 
-        return ModelCreator.MODEL_TRAINER_CONSTRUCTORS[MLModelType(model_type)]()
+        if MLModelType(pj["model"]) == MLModelType.XGB_QUANTILE:
+            return ModelCreator.MODEL_TRAINER_CONSTRUCTORS[MLModelType(pj["model"])](
+                tuple(pj["quantiles"])
```

Comment:
Why do we need a tuple here? If you really need a tuple internally I would suggest to do the conversion in the constructor.

Reply:
This is because of sklearn requirements, they want inputs in the __init__ to be imutable and not be muted in the __init__ itself.